### ISSUE TITLE
Browser Compatibility issues. 

### ISSUE THREAD

**Issue by Ra-Sedgwick** [AzureAD/microsoft-authentication-library-for-js#336]
My app is currently on v0.1.1,  using session storage and working in every browser but edge and safari which I have sla's to support. 

Safari gets caught in a infinite loop between my app and login.microsoft without ever stopping on one page. 
 
On Edge after submitting credentials and being redirected back to my site the error "user_login_error|User login is required" is being thrown by acquireTokenSilent and no id_token is present in session Storage. 

I’ve attempted the following to resolve the issue.

- Updating msal.js. 
   - I’ve tried several versions, 0.1.5, 0.1.6, 0.1.7
   - The same issue persist and now occurs in IE11 as well. 

![faillogin1](https://user-images.githubusercontent.com/10386425/41866568-8a3d1e70-787e-11e8-960d-2157113499dd.png)

![faillogin2](https://user-images.githubusercontent.com/10386425/41866798-3b5b5ed8-787f-11e8-8020-5b763b6c6365.png)

- [Known Issues in Edge](https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Known-issues-on-Edge#issues)
    -  Storing id_token in local storage instead of session.
    -  Adding app and login.microsoft.com urls to trusted sites. 

I've been stuck on this for days any assistance is appreciated..

**Comment by nehaagrawal** [comment]
@Ra-Sedgwick This is an existing issue with the Edge and IE browsers. We have documented this in our ADAL library but this applies here as well. Currently, we only have the work arounds listed here: https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Known-issues-on-Edge#issues

I have never seen issue on safari. Is it possible for you to share your repo or a sample project to reproduce this issue? Can you please tell me if you are seeing this issue in just local or live/production site?

**Comment by Ra-Sedgwick** [comment]
This is happening on localhost and in higher level environments. To clarify when I updated from 0.1.1 to 0.1.6 the safari issues resolved any my IE/Edge issues began. I will try to reproduce the issue in a sample app.  Thank you.

**Comment by nehaagrawal** [comment]
We have known issues with IE/Edge browser due to Security zone issue during redirect . We are currently working on a fix and it will be released sometime in future. The workaround to switch to localStorage may not work on all versions of IE/Edge. Did you try loginPopup() instead of loginRedirect() with the workaround settings?

**Comment by navyasric** [comment]
Thanks for the feedback. We will work on documenting this in Wiki of MSAL as well. Closing this issue as it is a known issue.

**Comment by Ra-Sedgwick** [comment]
I believe i found the cause of my issue, which seems to be a problem between IE11 and Aurelias DI library.

I tried resolving my authService( which contained all my msal code) class in main.js using aurelia.continer.get(authService) which resulted in authService.isAuthenticated() always returning false in IE.

```
  public isAuthenticated(): boolean {
    let user = this.userAgentApplication.getUser();
    return user !== null;
  }
```

I deleted all of my login logic from main.js, now handling all of it in the routers navigation pipeline. 

The class that configures my router imports the authService using @inject which worked in all browsers.
